channels/management/commands/backpopulate_missing_posts.py

```diff
@@ -0,0 +1,63 @@
+"""Management command for creating missing Post objects"""
+from django.conf import settings
+from django.contrib.auth.models import User
+from django.core.management import BaseCommand
+from django.db import transaction
+
+from channels.api import Api
+from channels.constants import LINK_TYPE_SELF, LINK_TYPE_LINK
+from channels.models import Channel, Post, Comment
+from channels.utils import get_or_create_link_meta
+from search.tasks import index_post_with_comments
+
+
+class Command(BaseCommand):
+    """Find and create missing Post objects (and associated comments) for reddit submissions"""
+
+    help = "Find and create missing Post objects for reddit submissions"
+
+    def handle(self, *args, **options):  # pylint:disable=too-many-locals
+        """Find and create missing Post objects for reddit submissions"""
+        total_missing = 0
+        api_client = Api(user=User.objects.get(username=settings.INDEXING_API_USERNAME))
+        for channel in Channel.objects.all():
+            subreddit = api_client.reddit.subreddit(channel.name)
+            reddit_posts = {reddit_post.id for reddit_post in subreddit.hot(limit=1000)}
+            django_posts = {
+                django_post.post_id for django_post in channel.post_set.all()
+            }
+            missing_posts = reddit_posts.difference(django_posts)
+            for post_id in missing_posts:
+                submission = api_client.get_post(post_id)
+                submission_type = (
+                    LINK_TYPE_LINK if hasattr(submission, "url") else LINK_TYPE_SELF
+                )
+                with transaction.atomic():
```

Comment:
Suggest that you wrap each attempt to repair a missing post in a `try` block and log the failures so that one single failure (i.e. a db write or failed call to embedly) doesn't fail the entire command